File: packages/dotnet/Tokid/TokidFactory.cs
using System;

namespace Tokid
{
    public sealed class TokidFactory
    {
        private readonly TokidRandomBytes _randomBytes;

        public string ProfileId { get; }
        public int Length { get; }

        public TokidFactory(string profileId = null, int? length = null)
            : this(profileId, length, null)
        {
        }

        internal TokidFactory(string profileId, int? length, TokidRandomBytes randomBytes)
        {
            ProfileId = profileId ?? TokidKernel.Default.DefaultProfileId;
            Length = length ?? TokidKernel.Default.Profile(ProfileId).Entropy.RecommendedLength;
            _randomBytes = randomBytes;
        }

        public string Generate(int? length = null, string format = "envelope")
        {
            var logical = TokidKernel.Default.GenerateTokid(ProfileId, length ?? Length, _randomBytes);
            return TokidKernel.Default.RenderTokid(logical, format);
        }

        public Tokid Parse(string value)
        {
            return TokidKernel.Default.ParseWithCodec(value, ProfileId, null);
        }

        public bool IsTokid(string value)
        {
            return TokidKernel.Default.ValidateTokid(value, ProfileId);
        }

        public string Prompt(object value)
        {
            return TokidSdk.ToPrompt(value, ProfileId);
        }

        public string Transport(object value)
        {
            return TokidSdk.ToTransport(value, ProfileId);
        }

        public string Envelope(object value)
        {
            return TokidSdk.ToEnvelope(value, ProfileId);
        }
    }
}

File: packages/java/src/main/java/io/tokid/TokidFactory.java
package io.tokid;

public final class TokidFactory {
  private final String profileId;
  private final int length;
  private final TokidRandomBytes randomBytes;

  public TokidFactory(String profileId, Integer length) {
    this(profileId, length, null);
  }

  TokidFactory(String profileId, Integer length, TokidRandomBytes randomBytes) {
    this.profileId = profileId == null ? TokidKernel.defaultKernel().defaultProfileId() : profileId;
    this.length =
        length == null
            ? TokidKernel.defaultKernel().profile(this.profileId).entropy().recommendedLength()
            : length;
    this.randomBytes = randomBytes;
  }

  public String profileId() {
    return profileId;
  }

  public int length() {
    return length;
  }

  public String generate() {
    return generate(null, null);
  }

  public String generate(Integer requestedLength, String format) {
    Tokid tokid =
        TokidKernel.defaultKernel()
            .generateTokid(profileId, requestedLength == null ? length : requestedLength, randomBytes);
    return TokidKernel.defaultKernel().renderTokid(tokid, format == null ? "envelope" : format);
  }

  public Tokid parse(String value) {
    return TokidKernel.defaultKernel().parseWithCodec(value, profileId, null);
  }

  public boolean isTokid(String value) {
    return TokidKernel.defaultKernel().validateTokid(value, profileId);
  }

  public String prompt(Object value) {
    return TokidSdk.toPrompt(value, profileId);
  }

  public String transport(Object value) {
    return TokidSdk.toTransport(value, profileId);
  }

  public String envelope(Object value) {
    return TokidSdk.toEnvelope(value, profileId);
  }
}
